client/verta/verta/_dataset_versioning/dataset.py

```diff
@@ -57,15 +57,54 @@ def _generate_default_name(cls):
 
     @classmethod
     def _get_proto_by_id(cls, conn, id):
-        raise NotImplementedError
+        Message = _DatasetService.GetDatasetById
+        msg = Message(id=id)
+        endpoint = "/api/v1/modeldb/dataset/getDatasetById"
+        response = conn.make_proto_request("GET", endpoint, params=msg)
+        return conn.maybe_proto_response(response, Message.Response).dataset
 
     @classmethod
     def _get_proto_by_name(cls, conn, name, workspace):
-        raise NotImplementedError
+        Message = _DatasetService.GetDatasetByName
+        msg = Message(name=name, workspace_name=workspace)
+        endpoint = "/api/v1/modeldb/dataset/getDatasetByName"
+        proto_response = conn.make_proto_request("GET", endpoint, params=msg)
+        response = conn.maybe_proto_response(proto_response, Message.Response)
+
+        if workspace is None or response.HasField("dataset_by_user"):
+            return response.dataset_by_user
+        elif hasattr(response, "shared_datasets") and response.shared_datasets:
+            return response.shared_datasets[0]
+        else:
+            return None
 
     @classmethod
     def _create_proto_internal(cls, conn, ctx, name, desc=None, tags=None, attrs=None, date_created=None, public_within_org=None):
```

Comment:
@convoliution I noticed that Dataset has a field "type":
https://github.com/VertaAI/modeldb/blob/master/protos/protos/public/modeldb/DatasetService.proto#L21
It was also set-able in the old Dataset:
https://github.com/VertaAI/modeldb/blob/master/client/verta/verta/client.py#L586
Should I add a field for it here, or just ignore it for now?

Reply:
We are 100% ignoring it for now and forever because it's being deprecated.